mm2src/lp_ordermatch.rs

```diff
@@ -229,91 +205,52 @@ async fn process_maker_order_updated(
 ///
 /// # Safety
 ///
 /// The function locks [`MmCtx::p2p_ctx`] and [`MmCtx::ordermatch_ctx`]
-async fn request_and_fill_orderbook(
-    ctx: &MmArc,
-    base: &str,
-    rel: &str,
-    asks_num: Option<usize>,
-    bids_num: Option<usize>,
-) -> Result<(), String> {
-    /*
-    // The function converts the given Vec<OrderInitialMessage> to Iter<Item = OrderbookItem>.
-    fn process_initial_messages(
-        initial_msgs: Vec<new_protocol::OrderInitialMessage>,
-    ) -> impl Iterator<Item = OrderbookItem> {
-        initial_msgs.into_iter().filter_map(
-            |new_protocol::OrderInitialMessage {
-                 initial_message,
-                 from_peer,
-                 update_messages,
-             }| match OrderbookItem::from_initial_msg(initial_message, update_messages, from_peer) {
-                Ok(order) => Some(order),
-                Err(e) => {
-                    log!("Error on parse OrderbookItem from initial message: "[e]);
-                    None
-                },
-            },
-        )
-    }
+async fn request_and_fill_orderbook(ctx: &MmArc, base: &str, rel: &str) -> Result<(), String> {
+    let ordermatch_ctx = OrdermatchContext::from_ctx(&ctx).unwrap();
+    let mut orderbook = ordermatch_ctx.orderbook.lock().await;
 
-    let get_orderbook = OrdermatchRequest::GetOrderbook {
+    let request = OrdermatchRequest::GetOrderbook {
         base: base.to_string(),
         rel: rel.to_string(),
-        asks_num,
-        bids_num,
     };
 
-    let responses =
-        try_s!(request_relays::<new_protocol::Orderbook>(ctx.clone(), P2PRequest::Ordermatch(get_orderbook)).await);
-
-    let mut asks = Vec::new();
-    let mut bids = Vec::new();
-    for (peer_id, peer_response) in responses {
-        match peer_response {
-            PeerDecodedResponse::Ok(orderbook) => {
-                asks.extend(process_initial_messages(orderbook.asks));
-                bids.extend(process_initial_messages(orderbook.bids));
-            },
-            PeerDecodedResponse::None => (),
-            PeerDecodedResponse::Err(e) => log!("Received error from peer " [peer_id] ": " [e]),
-        }
-    }
-
-    if let Some(n) = asks_num {
-        // the best asks are with the lowest prices (from lowest to highest prices)
-        asks.sort_by(|x, y| x.price.cmp(&y.price));
-        // truncate excess asks
-        asks.truncate(n)
-    }
-    if let Some(n) = bids_num {
-        // the best bids are with the highest prices (from highest to lowest prices)
-        bids.sort_by(|x, y| y.price.cmp(&x.price));
-        // truncate excess bids
-        bids.truncate(n)
-    }
-
-    let ordermatch_ctx = OrdermatchContext::from_ctx(&ctx).unwrap();
-    let mut orderbook = ordermatch_ctx.orderbook.lock().await;
+    let response = try_s!(request_any_relay::<GetOrderbookRes>(ctx.clone(), P2PRequest::Ordermatch(request)).await);
```

Comment:
Please note I changed calling `request_relays()` to `request_any_relay`, because now we request the whole orderbook.